packages/dashboard/src/components/iot-dashboard/iot-dashboard-dynamic-widget/iot-dashboard-dynamic-widget.tsx

```diff
@@ -0,0 +1,126 @@
+import { TimeQuery, TimeSeriesData, TimeSeriesDataRequest } from '@iot-app-kit/core';
+import { Component, Prop, h } from '@stencil/core';
+import {
+  Annotations,
+  LabelsConfig,
+  Trend,
+  MovementConfig,
+  ScaleConfig,
+  LayoutConfig,
+  Axis,
+  LegendConfig,
+  ChartConfig,
+  AlarmsConfig,
+  MessageOverrides,
+  MinimalSizeConfig,
+  MinimalViewPortConfig,
+} from '@synchro-charts/core';
+
+import xIcon from './icon-x.svg';
+
+type DynamicWidgetProps = ChartConfig & { queries: TimeQuery<TimeSeriesData[], TimeSeriesDataRequest>[] } & {
+  trends: Trend[] | undefined;
+} & { alarms: AlarmsConfig | undefined } & { labelsConfig: LabelsConfig | undefined } & { gestures: boolean };
+
+@Component({
+  tag: 'iot-dashboard-dynamic-widget',
+  styleUrl: 'iot-dashboard-dynamic-widget.css',
+})
+export class IotDynamicWidget implements ChartConfig {
+  @Prop() componentTag!: string;
+
+  @Prop() widgetId!: string;
+  @Prop() viewport: MinimalViewPortConfig;
+  @Prop() queries: TimeQuery<TimeSeriesData[], TimeSeriesDataRequest>[];
+  @Prop() movement?: MovementConfig;
+  @Prop() scale?: ScaleConfig;
+  @Prop() layout?: LayoutConfig;
+  @Prop() legend?: LegendConfig;
+  @Prop() annotations?: Annotations;
+  @Prop() axis?: Axis.Options;
+  @Prop() messageOverrides?: MessageOverrides;
+  @Prop() size?: MinimalSizeConfig;
+  @Prop() trends?: Trend[];
+  @Prop() alarms?: AlarmsConfig;
+  @Prop() gestures?: boolean;
+  @Prop() labelsConfig?: LabelsConfig;
+
+  @Prop() readOnly?: boolean;
+  @Prop() isEditing?: boolean;
+
+  /** Error Messaging */
+  @Prop() invalidTagErrorHeader!: string;
+  @Prop() invalidTagErrorSubheader!: string;
+
+  // A set of all componentTags which have caused errors.
+  // We want to keep track so that we don't call many errors for the same component tag.
+  private erroredTags: Set<string> = new Set();
+
+  /**
+   * returns the web-component tag to mount
+   *
+   * This should eventually be removed, once we refactor `status-chart`
+   * to be a `status-timeline`
+   */
+  get getComponentTag() {
+    // if (this.componentTag === 'monitor-status-timeline') {
```

Comment:
Is there a purpose for this method at all? I think we can just simply reference `componentTag`. This looks like a left over from some SiteWise Monitor specific oddities around migrating different component  tags